### PR TITLE
Remove private QLockFile implementation

### PR DESCRIPTION
**Pull request by shenlebantongying**
`<QLockFile>` is part of Qt since 5.1.

## To test

* Start GD
* Type `goldendict <word>`

* Close GD, then restart
* Type `goldendict <word>`

The `lockFile.open` is gone because this implementation detail is handled by QLockFile already.

fix https://github.com/xiaoyifang/goldendict-ng/issues/513
related https://github.com/xiaoyifang/goldendict-ng/issues/504

### CODE REVIEW

**Comment by xiaoyifang** [comment]
Another test result?:

Type goldendict <word1>

Type goldendict <word2>

**Comment by shenlebantongying** [comment]
It seems the old code doesn't delete the file, and rely on other things to check if it can lock it.

If you opened old version of GD, then locking won't work.

The `QLockFile::unlock` will also delete the file.

I have changed the lock file name a bit to avoid this problem.

**Comment by xiaoyifang** [comment]
> If you opened old version of GD, then locking won't work.

I see that you change a different lockname ,does this mean both the old gd and gd-ng can open together

**Comment by shenlebantongying** [comment]
Yes, it is pure based on if that file can be locked, but why someone would do this 😅

And how would they know this detail that between certain two versions, they can be opened together.

Beside this, the single application & messaging has been working constantly on my testings.

File: thirdparty/qtsingleapplication/src/qtlockedfile_unix.cpp [hunk 1]
```diff
@@ -1,119 +0,0 @@
...
-    if (m_lock_mode != NoLock)
-        unlock();
-
-    struct flock fl;
-    fl.l_whence = SEEK_SET;
-    fl.l_start = 0;
-    fl.l_len = 0;
-    fl.l_type = (mode == ReadLock) ? F_RDLCK : F_WRLCK;
-    int cmd = block ? F_SETLKW : F_SETLK;
-    int ret = fcntl(handle(), cmd, &fl);
```
Comment (shenlebantongying) [inline thirdparty/qtsingleapplication/src/qtlockedfile_unix.cpp:74]: The original implmention on Unix is adding F_RDLCK flag to the file. If the file doesn't exist, one will be created.

And the `<QLockFile>` will just delete it after use.

https://doc.qt.io/qt-6/qlockfile.html#unlock

**Comment by xiaoyifang** [comment]
> And how would they know this detail that between certain two versions, they can be opened together.

I think even can be opened together ,it should be ok .
This is fine

**Review by xiaoyifang — APPROVED** [review]

**Pull request merged**